src/stores/issue-page.tsx

```diff
@@ -0,0 +1,201 @@
+import { createStore as reduxCreateStore, Store as ReduxStore } from 'redux'
+
+export type IssuePageStore = ReduxStore<IssuePageState, IssuePageAction>
+
+const emptyState: IssuePageState = {
+  params: { state: 'checking' },
+  currentUser: { loaded: false },
+  issueState: { loading: true },
+  actionInProgress: null,
+  error: null,
+}
+
+function reducer(initialState: IssuePageState = emptyState, action: IssuePageAction): IssuePageState {
+  switch (action.type) {
+    case 'PAGE_LOAD': {
+      const params = new URLSearchParams(action.payload.search)
+
+      const site = params.get('site')
+      if (!site) {
+        return {
+          ...initialState,
+          params: {
+            state: 'not ok',
+            error: 'query param `site` is required',
+          },
+        }
+      }
+
+      const issueId = parseInt(params.get('id')!, 10)
+      if (!issueId) {
+        return {
+          ...initialState,
+          params: {
+            state: 'not ok',
+            error: 'query param `issueId`, an integer, is required',
+          },
+        }
+      }
+
+      return {
+        ...initialState,
+        params: {
+          state: 'ok',
+          params: { site, issueId },
+        },
+      }
+    }
+
+    case 'CURRENT_USER_LOAD_SUCCESS':
+      return {
+        ...initialState,
+        currentUser: {
+          loaded: true,
+          user: action.payload.user,
+        },
+      }
+
+    case 'INITIAL_ISSUE_LOAD_SUCCESS':
+      return {
+        ...initialState,
+        issueState: {
+          loading: false,
+          issue: action.payload.issue,
+        },
+      }
+
+    // Optimistically update the issue with the timeline that would result from the action.
+    // Set the actionInProgress with the prior state and the action so that this can be rolled back on an error.
+    case 'CHANGE_STATUS_INITIATE': {
```

Comment:
minor - Looks like lots of similar code between CHANGE_STATUS_INITIATE and POST_COMMENT_INITIATE. Some helper functions might make the processing steps more clearer.